jvm-libs/generic/extensions/kotlin/src/main/kotlin/net/consensys/TypingsExtensions.kt

```diff
@@ -78,6 +78,14 @@ fun ULongRange.intersection(other: ULongRange): ULongRange {
   }
 }
 
+fun ULong.minusCoercingUnderflow(other: ULong): ULong {
+  return if (this > other) {
+    this - other
+  } else {
+    0UL
+  }
+}
```

Comment:
one-liner suggestion: `return this.coerceAtLeast(other).minus(other)`

Reply:
I prefer to keep current implementation as it's easier to read IMO